### PR TITLE
[consensus] Make coinbase maturity a network parameter instead of an app-wide constant

### PR DESCRIPTION
**Pull request by kallewoof**
The coinbase maturity is a constant in `consensus/consensus.h` but ideally should be a parameter on a per network basis. Although all networks right now have the same coinbase maturity, there is no reason why this should be the case forever.

### CODE REVIEW

File: src/test/test_bitcoin.cpp [hunk 1]
```diff
@@ -100,7 +99,8 @@ TestChain100Setup::TestChain100Setup() : TestingSetup(CBaseChainParams::REGTEST)
     // Generate a 100-block chain:
     coinbaseKey.MakeNewKey(true);
     CScript scriptPubKey = CScript() <<  ToByteVector(coinbaseKey.GetPubKey()) << OP_CHECKSIG;
-    for (int i = 0; i < COINBASE_MATURITY; i++)
+    int coinbaseMaturity = Params().GetConsensus().coinbaseMaturity;
```
Comment (dcousens) [inline src/test/test_bitcoin.cpp]: `const`?

**Comment by dcousens** [comment]
concept ACK

File: src/test/test_bitcoin.cpp (hunk 1 above)
Comment (kallewoof) [inline src/test/test_bitcoin.cpp]: You mean `const int coinbaseMaturity = [...]`? Sure thing.

**Comment by NicolasDorier** [comment]
ACK 90bf56f

**Comment by theuni** [comment]
concept ACK, but please don't mix in all of the include/whitespace changes.

Surely that header is still needed in most cases, no? Just because it's indirectly include already doesn't mean the explicit include should be removed.

**Comment by kallewoof** [comment]
@theuni Sorry about whitespace. Fixed. I went through and restored the include in the files that were using other values from `consensus/consensus.h`.

**Comment by morcos** [comment]
Can someone explain the rationale for this change?

I don't see any reason to over generalize the code unless we anticipate wanting to take advantage of it and I feel the code is slightly cleaner and clearer as existing.

**Comment by TheBlueMatt** [comment]
I generally always thought of the consensus params stuff as parameters which change between networks. Unless we want a testnet with a different parameter here I'm not sure why bother?

**Comment by kallewoof** [comment]
It never made sense to me that it was not a part of the network parameters, even if they happen to be the same for all 3 network. (Personally I don't see why regtest requires a 100 coinbase, but that's a separate topic, I guess.)

**Comment by luke-jr** [comment]
NACK, I agree with @morcos. There's no reason to have this specific constant as a network parameter unless we are planning to make every single constant such also.

**Comment by kallewoof** [comment]
I guess you're right that it doesn't make a lot of sense and is easy enough to change if there ever is a need for a new network with a different coin base. Closing. Thanks for the feedback! :)

**Pull request closed without merging**